dags/oss_know/libs/zulip/crawl_stream.py

```diff
@@ -0,0 +1,64 @@
+import time
+import datetime
+from random import randint
+from dags.oss_know.libs.util.base import get_opensearch_client
+from oss_know.libs.util.log import logger
+from oss_know.libs.util.zulip import get_data_from_opensearch
+from oss_know.libs.base_dict.opensearch_index import OPENSEARCH_ZULIP_STREAM
+from oss_know.libs.util.opensearch_api import OpensearchAPI
+
+
+def crawl_zulip_stream(client, owner, repo, site, opensearch_conn_info):
+    max_try = 20
+    while(True):
+        flag = 1
+        try:
+            result = client.get_streams()
+        except Exception as e:
+            flag = 0
+        if flag and result["result"] == "success":
+            result = result["streams"]
+            break
+        time.sleep(randint(10, 30))
+        max_try -= 1
+        if max_try == 0:
+            logger.info("Fail to crawl zulip streams data.")
+            if flag:
+                logger.info(result["msg"])
+            else:
+                logger.info(e, exc_info=1)
+            return
+    opensearch_data = get_data_from_opensearch(OPENSEARCH_ZULIP_STREAM, owner, repo, opensearch_conn_info)
+    opensearch_client = get_opensearch_client(opensearch_conn_info=opensearch_conn_info)
+    opensearch_api = OpensearchAPI()
+
+    stream_exit = set()
```

Comment:
Seems like the `stream_exit` is used to check whether the data is inserted into opensearch or not. Then perhaps a `dict` is a better solution. Since `set` has to keep all elements unique, which takes much more effort in the background. 
Checking the existence of a key in `dict`(by hash algorithm) cost much less than checking item in list or set.

Reply:
The checks is to make sure just insert 'updated part' of the streams' data, on the solution perspective, maybe we can take these steps:
1. Init the zulip data(get all data at one time)
2. Set some check points(by date, typically)
3. Only get the new data which is later than the check point in the next round(so that we can make much less requests when fetching data)
4. Keep repeating the step 2 and step3